pkg/validation/internal/annotations.go

```diff
@@ -0,0 +1,61 @@
+package internal
+
+import (
+	"fmt"
+	"strings"
+
+	v1 "github.com/operator-framework/api/pkg/operators/v1"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/api/pkg/validation/errors"
+)
+
+// CaseSensitiveAnnotationKeySet is a set of annotation keys that are case sensitive
+// and can be used for validation purposes. The key is always lowercase and the value
+// contains the expected case sensitive string. This may not be an exhaustive list.
+var CaseSensitiveAnnotationKeySet = map[string]string{
+
+	strings.ToLower(v1.OperatorGroupAnnotationKey):             v1.OperatorGroupAnnotationKey,
+	strings.ToLower(v1.OperatorGroupNamespaceAnnotationKey):    v1.OperatorGroupNamespaceAnnotationKey,
+	strings.ToLower(v1.OperatorGroupTargetsAnnotationKey):      v1.OperatorGroupTargetsAnnotationKey,
+	strings.ToLower(v1.OperatorGroupProvidedAPIsAnnotationKey): v1.OperatorGroupProvidedAPIsAnnotationKey,
+	strings.ToLower(v1alpha1.SkipRangeAnnotationKey):           v1alpha1.SkipRangeAnnotationKey,
+
+	// below are duplicate values of known keys
+
+	// strings.ToLower(v1alpha1.OperatorGroupNamespaceAnnotationKey):    v1alpha1.OperatorGroupNamespaceAnnotationKey,
+	// strings.ToLower(v1alpha2.OperatorGroupAnnotationKey):             v1alpha2.OperatorGroupAnnotationKey,
+	// strings.ToLower(v1alpha2.OperatorGroupNamespaceAnnotationKey):    v1alpha2.OperatorGroupNamespaceAnnotationKey,
+	// strings.ToLower(v1alpha2.OperatorGroupTargetsAnnotationKey):      v1alpha2.OperatorGroupTargetsAnnotationKey,
+	// strings.ToLower(v1alpha2.OperatorGroupProvidedAPIsAnnotationKey): v1alpha2.OperatorGroupProvidedAPIsAnnotationKey,
+}
+
+/*
+ValidateAnnotationNames will check annotation keys to ensure they are using
+proper case. Uses CaseSensitiveAnnotationKeySet as a source for keys
+which are known to be case sensitive. This function can be used anywhere
+annotations need to be checked for case sensitivity.
+
+Arguments
+
+• annotations: annotations map usually obtained from ObjectMeta.GetAnnotations()
+
+• value: is the field or file that caused an error or warning
+
+Returns
+
+Any errors that may have been detected with the annotation keys provided
```

Comment:
Non-blocking nit: I am unfamiliar with the format, but based on the Arguments above should it match:
```suggestion
• errs: Any errors that may have been detected with the annotation keys provided
```

Reply:
accepted suggested change